server/meca/export.js

```diff
@@ -9,20 +9,33 @@ const archiveGenerator = require('./file-generators/archive')
 
 const upload = require('./services/upload')
 
-async function generate(manuscript, content, clientIp) {
-  return archiveGenerator({
+async function generate(manuscript, getContent, clientIp) {
+  let content = ''
+  const supportingFiles = {}
+
+  manuscript.files.forEach(file => {
+    if (file.type === 'MANUSCRIPT_SOURCE') {
+      content = getContent(file)
+    } else if (file.type === 'SUPPORTING_FILE') {
+      supportingFiles[file.filename] = getContent(file)
+    }
+  })
+
+  const manditoryFiles = {
     'article.xml': articleGenerator(manuscript),
     'cover_letter.html': coverLetterGenerator(manuscript),
     'disclosure.pdf': disclosureGenerator(manuscript, clientIp),
     'manifest.xml': manifestGenerator(manuscript.files),
     'manuscript.pdf': content,
     'transfer.xml': transferGenerator(''), // auth code not currently used
-  })
+  }
+
+  return archiveGenerator({ ...manditoryFiles, ...supportingFiles })
```

Comment:
Depending on the name of the supporting file, it could overwrite the manditory file.

e.g. user uploads a supporting file called `manifest.xml`

We may want to throw an error if this happens.

Reply:
Fair point